src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -45,11 +48,16 @@ public void OnBeginRequest(HttpContext httpContext, long timestamp)
                 this.client.Initialize(requestTelemetry);
                 requestTelemetry.Start(timestamp);
                 httpContext.Features.Set(requestTelemetry);
-
                 IHeaderDictionary responseHeaders = httpContext.Response?.Headers;
-                if (responseHeaders != null && !string.IsNullOrEmpty(requestTelemetry.Context.InstrumentationKey) && !responseHeaders.ContainsKey(RequestResponseHeaders.TargetInstrumentationKeyHeader))
+                if (responseHeaders != null &&
+                    !string.IsNullOrEmpty(requestTelemetry.Context.InstrumentationKey) &&
+                    (!responseHeaders.ContainsKey(RequestResponseHeaders.RequestContextHeader) || HttpHeadersUtilities.ContainsRequestContextKeyValue(responseHeaders, RequestResponseHeaders.RequestContextTargetKey)))
```

Comment:
The current implementation is more expensive in the sense of dealing different types of headers. As before, the header's we are interested is a dictionary and now it's single header contains sets of key value pairs.

|           | Key              	| Value                        	
|------- |------------------	|------------------------------	
|before | x-request-source 	| http://www.source.com        	
|after| context-header 	| source=http://www.source.com, blala=balabala 	

And that is why we build the utility to help us getting the value out.
With regard why we change the header format, I think that is off the topic of this specific change.